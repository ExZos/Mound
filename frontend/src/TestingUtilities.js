import GeneralComponent from './components/GeneralComponent';

class TestingUtilities extends GeneralComponent {
  addUserSessionItem(userID, spaceID, pollID=undefined) {
    const user = {
      id: userID,
      name: 'user' + userID,
      space: spaceID,
      space_name: 'space' + spaceID,
      space_status: true,
      poll: pollID
    };

    this.addToSessionArrayItem('users', user);
  }
}

export default TestingUtilities;
