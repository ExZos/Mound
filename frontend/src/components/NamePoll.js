import React from 'react';

import GeneralComponent from './GeneralComponent';
import { server, api } from '../server';
import '../styles/namePoll.css';

class NamePoll extends GeneralComponent {
  constructor(props) {
    super(props);

    this.state = ({
      user: {
        id: this.props.userID,
        space: this.props.spaceID,
        name: ''
      }
    });
  }

  addNamePoll = async () => {
    const poll = {
      space: this.state.user.space,
      user: this.state.user.id,
      name: this.state.user.name
    };

    try {
      const res = await server.post(api.createNameRelatedPoll, poll)

      let user = this.state.user;
      user.poll = res.data.id;

      this.addToSessionArrayItem('users', user);

      this.setState({
        user: {
          id: this.props.userID,
          space: this.props.spaceID,
          name: ''
        },
        poll: res.data
      });
    } catch (e) {
      // TODO: render error component
    }
  }

  render() {
    return (
      <div id="namePoll">
        <form id="addNamePoll" onSubmit={this.handleFormSubmit}>
          <input type="text" name="name" placeholder="Type a user name..."
            value={this.state.user.name}
            onChange={(e) => this.handleInputChange(e, 'user')}
          />

          <button tabIndex="-1" onClick={this.addNamePoll}>REQUEST</button>
        </form>
      </div>
    );
  }
}

export default NamePoll;
