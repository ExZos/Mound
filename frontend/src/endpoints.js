export const endpoints = {
  messages: '/api/messages/',
  spaces: '/api/spaces/',
  users: '/api/users/',
  poll: '/api/polls/',
  votes: '/api/votes/',
  getSpaceByName: '/space/getByName/',
  getUserCountInSpaceForUser: '/space/getUserCountForUser/',
  getUserByName: '/user/getByName/',
  getUsersInSpace: '/user/getInSpace/',
  getUserInSpaceByName: '/user/getInSpaceByName/',
  getUsersInSpaceExceptName: '/user/getInSpaceExceptName/',
  createUserNApproveSpace: 'user/createNApproveSpace/',
  getMessagesInSpace: '/message/getInSpace/',
  getPendingPollsInSpace: '/poll/getPendingInSpace/',
  getPendingUnvotedPollsForUser: '/poll/getPendingUnvotedForUser/',
  getPendingJoinPollInSpaceByName: 'poll/getPendingJoinInSpaceByName/',
  getJoinPollResults: '/poll/getJoinResults/',
  createNameRelatedPoll: '/poll/createNameRelated/',
  getVotesForPoll: 'vote/getForPoll/',
  getPositiveVotesForPoll: 'vote/getPositivesForPoll/',
  createVoteNUpdatePoll: 'vote/createNUpdatePoll/',
}
