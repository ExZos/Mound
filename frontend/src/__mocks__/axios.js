import { endpoints as api } from '../endpoints';

const axios = {
  create: () => axios,
  get: (url) => {
    switch(url) {
      case api.getSpaceByName + "Headon's Floor/":
        return Promise.resolve({
          data: {
            id: 1,
            name: "Headon's Floor",
            status: true
          }
        });
        break;

      default:
        return Promise.reject();
    }
  }
};

export default axios;
