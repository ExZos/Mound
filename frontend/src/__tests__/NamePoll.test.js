import React from 'react';
import { shallow, mount, render } from 'enzyme';
import axios from 'axios';
import { createBrowserHistory } from 'history';

import NamePoll from '../components/NamePoll';
import TestingUtilities from '../TestingUtilities';
import { endpoints as api } from '../endpoints';

const history = createBrowserHistory();
const tutils = new TestingUtilities();

describe('NamePoll', () => {
  it('should render name correctly with props', async () => {
    const spy = jest.spyOn(axios, 'get');

    const component = await shallow(<NamePoll spaceID={1} userID={1} />);
    const form = component.find('#addNamePoll');

    expect(component).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
    expect(form).toHaveLength(1);
  });

  it('should render pending poll correctly with props', async () => {
    const spy = jest.spyOn(axios, 'get');

    const component = await shallow(<NamePoll spaceID={2} userID={2} />);
    const statusStatement = component.find('div.statusStatement');
    const progressStatement = component.find('div.progressStatement');
    const resultingStatement = component.find('div.resultingStatement');

    expect(component).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
    expect(statusStatement).toHaveLength(1);
    expect(statusStatement.childAt(1).text()).toContain('pending');
    expect(progressStatement).toHaveLength(1);
    expect(progressStatement.childAt(1).text()).toEqual('5');
    expect(progressStatement.childAt(3).text()).toEqual('10');
    expect(resultingStatement).toHaveLength(1);
    expect(resultingStatement.text()).toContain('not yet voted');
  });

  it('should render approved poll correctly with props', async () => {
    const spy = jest.spyOn(axios, 'get');

    const component = await shallow(<NamePoll spaceID={3} userID={3} />);
    const statusStatement = component.find('div.statusStatement');
    const progressStatement = component.find('div.progressStatement');
    const resultingStatement = component.find('div.resultingStatement');

    expect(component).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
    expect(statusStatement).toHaveLength(1);
    expect(statusStatement.childAt(1).text()).toContain('pending');
    expect(progressStatement).toHaveLength(1);
    expect(progressStatement.childAt(1).text()).toEqual('3');
    expect(progressStatement.childAt(3).text()).toEqual('3');
    expect(resultingStatement).toHaveLength(1);
    expect(resultingStatement.text()).toContain('approved');
  });

  it('should render rejected poll correctly with props', async () => {
    const spy = jest.spyOn(axios, 'get');

    const component = await shallow(<NamePoll spaceID={4} userID={4} />);
    const statusStatement = component.find('div.statusStatement');
    const progressStatement = component.find('div.progressStatement');
    const resultingStatement = component.find('div.resultingStatement');

    expect(component).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
    expect(statusStatement).toHaveLength(1);
    expect(statusStatement.childAt(1).text()).toContain('rejected');
    expect(progressStatement).toHaveLength(1);
    expect(progressStatement.childAt(1).text()).toEqual('2');
    expect(progressStatement.childAt(3).text()).toEqual('7');
    expect(resultingStatement).toHaveLength(1);
    expect(resultingStatement.text()).toContain('rejected');
  });
});
