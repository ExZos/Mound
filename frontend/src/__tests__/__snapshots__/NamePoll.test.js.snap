// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamePoll should render approved poll correctly with props 1`] = `
<div
  id="namePoll"
>
  <div
    className="statusStatement"
  >
    Status: 
    pending
  </div>
  <div
    className="progressStatement"
  >
    Progress: 
    3
    /
    3
  </div>
  <br />
  <div
    className="resultingStatement"
  >
    <div>
      The required votes has been met.
      <br />
      Your request will be approved shortly.
    </div>
  </div>
</div>
`;

exports[`NamePoll should render name correctly with props 1`] = `
<div
  id="namePoll"
>
  <form
    id="addNamePoll"
    onSubmit={[Function]}
  >
    <div
      className="textFieldWButton"
    >
      <WithStyles(ForwardRef(TextField))
        label="Type a user name..."
        name="name"
        onChange={[Function]}
        size="small"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        size="small"
        tabIndex="-1"
        variant="outlined"
      >
        REQUEST
      </WithStyles(ForwardRef(Button))>
    </div>
  </form>
</div>
`;

exports[`NamePoll should render pending poll correctly with props 1`] = `
<div
  id="namePoll"
>
  <div
    className="statusStatement"
  >
    Status: 
    pending
  </div>
  <div
    className="progressStatement"
  >
    Progress: 
    5
    /
    10
  </div>
  <br />
  <div
    className="resultingStatement"
  >
    <div>
      Some users have not yet voted.
    </div>
  </div>
</div>
`;

exports[`NamePoll should render rejected poll correctly with props 1`] = `
<div
  id="namePoll"
>
  <div
    className="statusStatement"
  >
    Status: 
    rejected
  </div>
  <div
    className="progressStatement"
  >
    Progress: 
    2
    /
    7
  </div>
  <br />
  <div
    className="resultingStatement"
  >
    <div>
      All users have voted.
      <br />
      The required votes has not been met.
      <br />
      Your request has been rejected.
    </div>
  </div>
</div>
`;
